schema/ref.go

```diff
@@ -80,8 +92,10 @@ func (r *RefSchema) Display() Display {
 	return r.DisplayValue
 }
 
-func (r *RefSchema) ApplyScope(scope Scope) {
-
+func (r *RefSchema) ApplyScope(scope Scope, namespace string) {
+	if namespace != r.ObjectNamespace {
+		return
+	}
```

Comment:
So the new `namespace` parameter is just a *check* against the receiver's namespace, and we silently return if it doesn't match? I find myself a bit confused, but perhaps it'll sink in ...

Reply:
I added a comment.

Reply:
Hmm. Maybe a function comment clarifying that you're "applying a scope within a namespace" and, in particular, that references in other namespaces are skipped? The new comment is better than nothing, but isn't very comprehensive.

Reply:
How about now?

Reply:
I'm not comfortable with this comment as amended, and part of that comes from the implementation which it is endeavoring to describe.  But, at this point, I would rather not hold up the merge of this PR while we all educate each other, so let's just get this merged, and we can thrash about later.